experimental/packages/exporter-logs-otlp-http/src/platform/node/OTLPLogExporter.ts

```diff
@@ -39,12 +39,13 @@ export class OTLPLogExporter
       timeoutMillis: getEnv().OTEL_EXPORTER_OTLP_LOGS_TIMEOUT,
       ...config,
     });
-    this.headers = {
-      ...this.headers,
-      ...baggageUtils.parseKeyPairsIntoRecord(
+    this.headers = Object.assign(
```

Comment:
Hmm, looks like the logs exporter use different way of merging the headers together, maybe we should align the other one (logs-protobuf) to also create a new object and use the spread operator instead.

Reply:
The problem with using the spread operator is that it will cause a type error: 

`Type '{ [x: string]: unknown; }' is not assignable to type 'Record<string, string>'`

That's caused by `config.headers` being defined as `Record<string, unknown>`, while `this.headers` expects `Record<string, string>`.


In other exporters, for example trace-http, there is is a hardcoded header

https://github.com/open-telemetry/opentelemetry-js/blob/f4b681d4f388ecedaecff03b7c95f460c97a8f6c/experimental/packages/exporter-trace-otlp-http/src/platform/node/OTLPTraceExporter.ts#L33-L35

which will ensure that the type of the new object is `Record<string, string>`

https://github.com/open-telemetry/opentelemetry-js/blob/f4b681d4f388ecedaecff03b7c95f460c97a8f6c/experimental/packages/exporter-trace-otlp-http/src/platform/node/OTLPTraceExporter.ts#L46-L52
(if we remove L48, we'll get the same type error)

This brings another question (unrelated to the scope of this PR): Why is no user agent specified for log exporters? As per this spec, all exporters should have it: https://opentelemetry.io/docs/specs/otel/protocol/exporter/#user-agent


I propose the following course of action:
1. keep the Object.assign in exporter-trace-otlp-http and exporter-trace-otlp-http to keep the compiler happy as it is just a cosmetic issue
2. create a new issue specifying the exporters which do not send the user agent header per [spec](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#user-agent)
3. solve the user agent issue in the subsequent PR which will also solve the cosmetic inconsistency issue by replacing `Object.assign` with using the spread operator everywhere.

Reply:
> I propose the following course of action:
> 
>     1. keep the Object.assign in exporter-trace-otlp-http and exporter-trace-otlp-http to keep the compiler happy as it is just a cosmetic issue
> 
>     2. create a new issue specifying the exporters which do not send the user agent header per [spec](https://opentelemetry.io/docs/specs/otel/protocol/exporter/#user-agent) [User agent header not set for log exporters #4397](https://github.com/open-telemetry/opentelemetry-js/issues/4397)
> 
>     3. solve the user agent issue in the subsequent PR which will also solve the cosmetic inconsistency issue by replacing `Object.assign` with using the spread operator everywhere.

ah, yes that makes sense, let's proceed like this then :+1: 
Thanks for opening the issue and looking into that :slightly_smiling_face: